src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -1873,10 +1898,149 @@ public void testEventStreamPolling() throws Exception {
     assertNotNull(eventList);
     assertThat(eventList, not(empty()));
 
-    final Event pullEvent = eventList.get(0);
-    final Event createEvent = eventList.get(1);
-    final Event startEvent = eventList.get(2);
-    eventStreamAssertions(pullEvent, createEvent, startEvent, container.id(), BUSYBOX_LATEST);
+    imagePullEventAssertions(eventList.get(0));
+
+    // create and start event assertions
+    containerEventAssertions(eventList.get(1), containerId, containerName,
+            "create", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(2), containerId, containerName,
+            "start", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(3), containerId, containerName,
+            "die", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(4), containerId, containerName,
+            "destroy", BUSYBOX_LATEST);
+  }
+
+  @Test(timeout = 10000)
+  public void testEventTypes() throws Exception {
+    requireDockerApiVersionAtLeast("1.22", "Event types");
+
+    final String volumeName = randomName();
+    final String containerName = randomName();
+    final String mountPath = "/anywhere";
+    final Volume volume = Volume.builder().name(volumeName).build();
+    final HostConfig hostConfig = HostConfig.builder()
+            .binds(Bind.from(volume).to(mountPath).build())
+            .build();
+    final ContainerConfig config = ContainerConfig.builder()
+            .image(BUSYBOX_LATEST)
+            .hostConfig(hostConfig)
+            .build();
+
+    // Wait once to clean our event "palette" of events from other tests
+    Thread.sleep(1000);
```

Comment:
I think I could do that. There is a case to avoid: when you are streaming events, if you ask for the next one and there is none, the thread will hang while it waits for more to come over the stream. But I can careful in my tests to only assert that there are no events when I have polled for events and the stream is already closed. Also, setting timeouts on the tests will help.